consumer.go

```diff
@@ -19,6 +19,7 @@ const (
 	deadLetterExchangeKind = amqp.ExchangeDirect
 	retryExchange          = "amq.direct"
 	retryExchangeKind      = amqp.ExchangeDirect
+	deferExchange          = "amq.direct"
```

Comment:
I'm pretty sure we can remove this list of constants since every rule (normal, retry, defer, deadLetter) will get a binding to `amq.direct` with `direct` kind.

Only 1 queue can get more bindings: the normal queue or defer if applicable

Reply:
And therefore, you don't need to send it as a parameter to setupQueue, since setupQueue will always create this binding, using the queue name as RK